ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1248,6 +1258,29 @@ private void prepareForSending() {
       }
     }
 
+    /**
+     * Resolve the reserved metadata id to be used for data chunks.
+     * A chunked upload will always have a metadata id as specified in the passed-in blob properties.
+     * A stitch upload will not have a metadata id as it is a metadata blob itself.
+     * All other uploads (i.e, regular uploads) should have a metadata id only for composite blobs.
+     * Note that this method assumes that all the validations regarding reserved metadata chunk id have been completed
+     * and a valid reserved metadata id is present for chunked uploads and composite blobs.
+     * @return the blob id to use as the reserved metadata id. Can be {@code null}.
+     */
+    private String resolveReservedMetadataIdForDataChunk() {
+      if (!routerConfig.routerReservedMetadataEnabled) {
+        return null;
+      }
+      if (options.isChunkUpload()) {
+        return passedInBlobProperties.getReservedMetadataBlobId();
+      }
+      if (!chunkFillingCompletedSuccessfully || (chunkFillingCompletedSuccessfully
```

Comment:
Can you embed comment for each condition in the code instead of in the function header?
I think "chunkFillingCompletedSuccessfully" means it's a data chunk.

What does this condition mean? Is this the last data chunk of the composite blob?
(chunkFillingCompletedSuccessfully && && getBlobSize() > routerConfig.routerMaxPutChunkSizeBytes)

Reply:
If we can have some wrap functions like IsDataChunk(), IsMetaChunk() etc, that'll be great.